components/raftstore-v2/src/operation/command/admin/merge/prepare.rs

```diff
@@ -350,6 +539,24 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         store_ctx: &mut StoreContext<EK, ER, T>,
         applied_index: u64,
     ) {
+        if self.merge_context().is_none() {
+            return;
+        }
+        // Check if there's a stale trim check. Ideally this should be implemented as a
+        // tick. But this is simpler.
+        if let Some(PrepareStatus::WaitForTrimStatus {
+            start_time, req, ..
+        }) = self
+            .merge_context()
+            .as_ref()
+            .and_then(|c| c.prepare_status.as_ref())
+            && req.is_some()
+            && start_time.saturating_elapsed() > TRIM_CHECK_TIMEOUT
```

Comment:
There should be a tick for retry, otherwise the memory may be leaked if there is an isolation and PD cancels merging.